Add a WebXR chess demo to Webkit.org demos

AdaRoseCannon opened the pull request:
#### 21aa0dc6b6a0d1812d7f8fd9b54df776ab8b42cc
<pre>
Add a WebXR chess demo to Webkit.org demos
<a href="https://bugs.webkit.org/show_bug.cgi?id=275240">https://bugs.webkit.org/show_bug.cgi?id=275240</a>

Reviewed by Aditya Keerthi.

Add demo to webkit.org which uses transient-pointer in WebXR on visionOS.

* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_decoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_decoder.wasm: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_encoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_wasm_wrapper.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/README.md: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/basis_transcoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/basis_transcoder.wasm: Added.
* Websites/webkit.org/demos/webxr-chess/assets/bake.hdr: Added.
* Websites/webkit.org/demos/webxr-chess/assets/baketiny.hdr: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chess-table.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chess.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessA.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessB.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessC.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/garden.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/garden4k.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/leaf_sprite_variants_bc.png: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/index.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/short-hand-standalone.min.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/short-hand-standalone.min.js.map: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/shorthand-patch.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/chess.js.license: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/chess.min.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/noise.js: Added.
* Websites/webkit.org/demos/webxr-chess/chess.css: Added.
* Websites/webkit.org/demos/webxr-chess/chess.jpg: Added.
* Websites/webkit.org/demos/webxr-chess/index.html: Added.

Canonical link: <a href="https://commits.webkit.org/279821@main">https://commits.webkit.org/279821@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f2b1efbbaa552a19a8dd0e133cdca52965a20737

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54652 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/34074 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7227 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57932 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/5385 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/41622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5393 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44271 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3633 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56747 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/41622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47337 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25395 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/41622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3525 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/41622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4885 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59522 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29890 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/5029 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51693 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31035 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47423 "Build is in progress. Recent messages:OS: Ventura (13.6.6), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/51078 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11974 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/32022 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30822 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Websites/webkit.org/demos/webxr-chess/index.html:
```diff
@@ -0,0 +1,157 @@
+                    <sh-geometry-cylinder id="leaf-geo" sweep-length="90" open="true" segments-radial="12" height="0.15" radius-bottom="0.1" radius-top="0.1" ></sh-geometry-cylinder>
+                    <sh-material-standard id="leaf-mat" roughness="0.8" metalness="0" map="#leaf-tex" transparent="true" backface-visibility="visible" alpha-to-coverage="true"></sh-material-standard>
+                    <!-- <sh-material-depth id="leaf-mat-depth" map="#leaf-tex" depth-packing="depth" alpha-hash="true" alpha-test="0.5"></sh-material-phong> -->
+                    <!-- <sh-mesh id="leafMesh" geometry="#leaf-geo" depth-material="#leaf-mat-depth" material="#leaf-mat" location="0 0.66 -0.65"></sh-mesh> -->
```
Any reason to keep these commented out lines?

AdaRoseCannon replied:
They are mostly things which I wasn't sure if I wanted to keep around so I will remove them.

pxlcoder commented on Websites/webkit.org/demos/webxr-chess/index.html:
```diff
@@ -0,0 +1,157 @@
+            <sh-material-phong emissive="#888888" environment-map="#hdrs" shininess="0" reflectivity="0.7" id="panewhite"></sh-material-phong>
+            <sh-material-standard color="#222222" roughness="0.7" id="fontblack"></sh-material-standard>
+            <sh-material-phong emissive="#333333" combine="mix" environment-map="#hdrs" shininess="0" reflectivity="0.4"id="buttonmaterial"></sh-material-phong>
+            <!-- <sh-material-standard color="darkgrey" id="buttonmaterial"></sh-material-standard> -->
```
Same question about the commented out line.

pxlcoder commented on Websites/webkit.org/demos/webxr-chess/index.html:
```diff
@@ -0,0 +1,157 @@
+    <article>
+        <h2>Chess Garden</h2>
+        <blockquote><p>When the game is over, the king and pawn go into the same box.</p><footer>—Proverb</footer></blockquote>
+        <p>This is an example of a chess game that uses WebXR. To play you need to view the page on a browser with WebXR capabilities on hardware with XR capabilities such as Safari on visionOS on Apple Vision Pro. In this versions the CPU player makes random but valid moves, it would probably be more engaging to integrate a Chess engine into the game so that it plays smarter.</p>
```
```suggestion
        <p>This is an example of a chess game that uses WebXR. To play you need to view the page on a browser with WebXR capabilities on hardware with XR capabilities such as Safari on visionOS on Apple Vision Pro. In this version the CPU player makes random but valid moves, it would probably be more engaging to integrate a Chess engine into the game so that it plays smarter.</p>
```

pxlcoder commented on Websites/webkit.org/demos/webxr-chess/index.html:
```diff
@@ -0,0 +1,157 @@
+                                <sh-use id="sessionstart" onclick="xrsession.enterVR()" class="auto-text-shape clickable" color="white" material="#buttonmaterial">Enter VR</sh-use>
+                                <sh-use id="sessionend" onclick="xrsession.endXR()" class="auto-text-shape clickable" color="white" material="#buttonmaterial">Leave garden</sh-use>
+                            </sh-group>
+                            <!-- <sh-text location="-0.3 0.025 0" font-size="0.05" anchor-y="bottom" anchor-x="left" color="black"><span id="fpscounter" slot="text"></span></sh-text> -->
```
Same question.

pxlcoder commented on Websites/webkit.org/demos/webxr-chess/chess.css:
```diff
@@ -0,0 +1,157 @@
+html,
+body {
+    height: 100%;
+    background-color: white;
```
Should we support dark mode?

AdaRoseCannon replied:
We can but it's not really about the text.

I also don't have a night version of the chess garden (although that would be neat for a future project).

AdaRoseCannon committed:
Add a WebXR chess demo to Webkit.org demos
https://bugs.webkit.org/show_bug.cgi?id=275240

Reviewed by Aditya Keerthi.

Add demo to webkit.org which uses transient-pointer in WebXR on visionOS.

* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_decoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_decoder.wasm: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_encoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/DRACO/draco_wasm_wrapper.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/README.md: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/basis_transcoder.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/KTX2/basis_transcoder.wasm: Added.
* Websites/webkit.org/demos/webxr-chess/assets/bake.hdr: Added.
* Websites/webkit.org/demos/webxr-chess/assets/baketiny.hdr: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chess-table.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chess.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessA.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessB.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/chessC.m4a: Added.
* Websites/webkit.org/demos/webxr-chess/assets/garden.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/garden4k.glb: Added.
* Websites/webkit.org/demos/webxr-chess/assets/leaf_sprite_variants_bc.png: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/index.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/short-hand-standalone.min.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/short-hand-standalone.min.js.map: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/shorthand-patch.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/chess.js.license: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/chess.min.js: Added.
* Websites/webkit.org/demos/webxr-chess/assets/scripts/thirdparty/noise.js: Added.
* Websites/webkit.org/demos/webxr-chess/chess.css: Added.
* Websites/webkit.org/demos/webxr-chess/chess.jpg: Added.
* Websites/webkit.org/demos/webxr-chess/index.html: Added.

Canonical link: https://commits.webkit.org/279821@main
